4_basic_assembly/2_bitwise_operations/2_practical_bit_games/answers/3_write_code/4.asm

```diff
@@ -0,0 +1,44 @@
+; Basic Assembly
+; ==============
+;
+; 4.  Rotation using shifts.
+; 
+;     Implement the ror instruction using the shift instructions. You may use any
+;     bitwise instruction for this task, except for the rotation instructions
+;     (ror,rol).
+

```

Comment:
@liara : This is a cool solution.
I think that you can do this even without a loop if you want to. Something like this:

```asm
mov      edx, eax
shr       edx, k
shl       eax, 32 - k
or        eax, edx
```
What do you think?

Reply:
I see what you mean. I guess I can put k and then 32-k in cl.